code/datums/diseases/advance/symptoms/skin.dm

```diff
@@ -181,3 +181,137 @@ BONUS
+
+/datum/symptom/skineggs/severityset(datum/disease/advance/A)
+	. = ..()
+	if(A.properties["resistance"] >= 10)
+		severity -= 1
+	if(A.properties["transmittability"] >= 12)
+		severity += 1
+	if(A.properties["transmittability"] >= 16)
+		severity += 1
+	if(A.properties["stealth"] >= 6)
+		severity += 1
+
+/datum/symptom/skineggs/Start(datum/disease/advance/A)
+	if(!..())
+		return
+	if(A.properties["resistance"] >= 10)
+		thresholds["bigheal"] = TRUE
+	if(A.properties["transmittability"] >= 12)
+		thresholds["alldisease"] = TRUE
+	if(A.properties["transmittability"] >= 16)
+		thresholds["eggsplosion"] = TRUE //Haha get it?
+	if(A.properties["stealth"] >= 6)
+		thresholds["sneaky"] = TRUE
+	if(A.properties["stage_rate"] >= 10)
+		symptom_delay_min -= 10
+		symptom_delay_max -= 20
+
+
+/datum/symptom/skineggs/Activate(datum/disease/advance/A)
+	if(!..())
+		return
+	var/mob/living/M = A.affected_mob
+	var/list/diseases = list(A)
+	switch(A.stage)
+		if(5)
+			if(thresholds["alldisease"])
+				for(var/datum/disease/D in M.diseases)
+					if((D.spread_flags & DISEASE_SPREAD_SPECIAL) || (D.spread_flags & DISEASE_SPREAD_NON_CONTAGIOUS) || (D.spread_flags & DISEASE_SPREAD_FALTERED))
+						continue
+					if(D == A)
+						continue
+					diseases += D
+			new /obj/item/reagent_containers/food/snacks/eggsac(diseases, thresholds["eggsplosion"], thresholds["sneaky"], thresholds["bigheal"])
+/obj/item/reagent_containers/food/snacks/eggsac
+	name = "Fleshy Egg Sac"
+	desc = "A small Egg Sac which appears to be made out of someone's flesh!"
+	customfoodfilling = 0 //Not Used For Filling
+	icon_state = "icons/obj/food/food/eggsac.dmi"
+	bitesize = 4
+	var/list/diseases = list()
+	var/sneakyegg
+	var/bigheal
+
+//Constructor
+/obj/item/reagent_containers/food/snacks/eggsac/New(var/list/disease, var/eggsplodes, var/sneaky, var/largeheal)
+	for(var/datum/disease/D in disease)
+		diseases += D
+	if(largeheal)
+		src.add_initial_reagents(list(/datum/reagent/medicine/bicaridine = 20, /datum/reagent/medicine/tricordrazine = 10))
+		src.reagents.add_reagent(/datum/reagent/blood, 10, diseases)
+		bigheal = TRUE
+	else
+		src.add_initial_reagents(list(/datum/reagent/medicine/bicaridine = 10, /datum/reagent/medicine/tricordrazine = 10))
+		src.reagents.add_reagent(/datum/reagent/blood, 15, diseases)
+	if(sneaky)
+		icon_state = "icons/obj/food/food/eggsac-sneaky.dmi"
+		sneakyegg = sneaky
+	if(eggsplodes)
+		addtimer(CALLBACK(src, .proc/eggsplode), 100 SECONDS)
+	if(LAZYLEN(diseases))
+		AddComponent(/datum/component/infective, diseases)
+
+
+/obj/item/reagent_containers/food/snacks/eggsac/proc/eggsplode()
+	var/eggcount = rand(4, 8)
+	var/i
+	for(i = 1, i<= eggcount, i++)
+		var/list/directions = GLOB.alldirs[i]
+		var/obj/item/I = new /obj/item/reagent_containers/food/snacks/fleshegg(src.diseases, src.sneakyegg, src.bigheal)
+		var/turf/thrown_at = get_ranged_target_turf(I, pick(directions), rand(2, 4))
```

Comment:
Why are you saving single index of `alldirs` to new list and the picking a single item from it?
```suggestion
		var/obj/item/I = new /obj/item/reagent_containers/food/snacks/fleshegg(diseases, sneakyegg, bigheal)
		var/turf/thrown_at = get_ranged_target_turf(I, GLOB.alldirs[i], rand(2, 4))
```

Reply:
That was not intentional